src/towncrier/_builder.py

```diff
@@ -181,6 +182,22 @@ def find_fragments(
     return content, fragment_files
 
 
+def get_underline_size(text: str) -> int:
+    """
+    Given `text` determine the underline size needed for the reStructuredText output.
+
+    Particularly helps determine if an extra underline is needed for wide characters like emojis.
+    """
+    underline_size: int = 0
+    for char in text:
+        if unicodedata.east_asian_width(char) in ("W", "F"):
```

Comment:
We also need a integration test.

If for example we have something like `if False:`
the existing tests will still pass with success. 

Please consider creating a test similar to this one, but in which a custom category is used, one that has unicode in the name.

In this way, we can make sure that this functionality will continue to work in future versions.

https://github.com/twisted/towncrier/blob/fc3a2556408bcdb0237e54c4ff0773db364c554a/src/towncrier/test/test_build.py#L1042-L1090